tools/testValidJson.ps1

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+<#
+    .SYNOPSIS
+        Get all JSON files recursively and test if they are valid by trying to import them.
+
+    .EXAMPLE
+        & $psEditor.GetEditorContext().CurrentFile.Path -WorkingDir '.\'
+#>
+
+# Input and expected output
+[OutputType([System.Void])]
+Param(
+    [Parameter()]
+    [ValidateScript({[System.IO.Directory]::Exists($_)})]
+    [string] $WorkingDir = '.\'
+)
+
+# PowerShell preferences
+$ErrorActionPreference = 'Stop'
+
+# Try to import all JSON files in the repo
+$TestValidJson = [PSCustomObject[]](
+    [System.IO.Directory]::GetFiles($WorkingDir,'*.json',[System.IO.SearchOption]::AllDirectories).ForEach{
+        [PSCustomObject]@{
+            'Path'        = [string] $_
+            'IsValidJson' = [bool]$(
+                Try {
+                    $null = ConvertFrom-Json -InputObject (Get-Content -Raw -Path $_) -AsHashtable
```

Comment:
Just curious, why not `Test-Json` and a simpler pipeline?

Reply:
I've experienced that `Test-Json` fails on valid JSON where `ConvertFrom-Json` doesn't, especially for PowerShell v7.4.

And that `Test-Json` behaves differently between PowerShell 7.3 and 7.4.

Something about Newtonsoft.Json vs. System.Text.Json? 🤔

Reply:
I really appreciate the idea to sanity check our JSON in CI, it's a great one. We should just go much simpler and check the snippets file. Trying to run this script locally after a build runs it against a million files under `node_modules` and `modules` etc., and it's really only the snippets that we need to check anyway.

Reply:
Ah. Sure, go ahead.

The idea was to:

1. Output what exact JSON files are not tested as valid, instead of just failing on the first one that is (could be multiple).
1. Create something generic I can use in other pipelines later too, thus a bit more than a oneliner.